sdk/storage/Azure.Storage.DataMovement.Files.Shares/tests/StartTransferUploadTests.cs

```diff
@@ -0,0 +1,677 @@
+
+                // Verify Upload
+                await uploadedBlobInfo[i].EventsRaised.AssertSingleCompletedCheck();
+                using (FileStream fileStream = File.OpenRead(uploadedBlobInfo[i].LocalPath))
+                {
+                    await DownloadAndAssertAsync(fileStream, uploadedBlobInfo[i].DestinationClient);
+                }
+            }
+        }
+
+        [RecordedTest]
+        public async Task LocalToShareFile()
+        {
+            // Arrange
+            await using DisposingShare testShare = await GetTestShareAsync();
+            using DisposingLocalDirectory localDirectory = DisposingLocalDirectory.GetTestDirectory();
+
+            await UploadShareFilesAndVerify(testShare.Share);
+        }
+
+        [RecordedTest]
+        public async Task LocalToShareFile_EventHandler()
+        {
+            AutoResetEvent InProgressWait = new AutoResetEvent(false);
+
+            bool progressSeen = false;
+            DataTransferOptions options = new DataTransferOptions();
+            options.TransferStatusChanged += (TransferStatusEventArgs args) =>
+            {
+                // Assert
+                if (args.TransferStatus.State == DataTransferState.InProgress)
+                {
+                    progressSeen = true;
+                }
+                return Task.CompletedTask;
+            };
+
+            // Arrange
+            await using DisposingShare testShare = await GetTestShareAsync();
+
+            List<DataTransferOptions> optionsList = new List<DataTransferOptions>() { options };
+            await UploadShareFilesAndVerify(
+                share: testShare.Share,
+                blobCount: optionsList.Count,
+                options: optionsList);
+
+            // Assert
+            Assert.IsTrue(progressSeen);
+        }
+
+        [RecordedTest]
+        public async Task LocalToShareFileSize_SmallChunk()
+        {
+            long fileSize = Constants.KB;
+            int waitTimeInSec = 25;
+            DataTransferOptions options = new DataTransferOptions()
+            {
+                InitialTransferSize = 100,
+                MaximumTransferChunkSize = 200,
+            };
+
+            // Arrange
+            await using DisposingShare testShare = await GetTestShareAsync();
+            List<DataTransferOptions> optionsList = new List<DataTransferOptions>() { options };
+
+            await UploadShareFilesAndVerify(
+                size: fileSize,
+                waitTimeInSec: waitTimeInSec,
+                share: testShare.Share,
+                options: optionsList);
+        }
+
+        [RecordedTest]
+        public async Task LocalToShareFile_Overwrite_Exists()
+        {
+            // Arrange
+            // Create source local file for checking, and source blob
+            await using DisposingShare testShare = await GetTestShareAsync();
+            using DisposingLocalDirectory testDirectory = DisposingLocalDirectory.GetTestDirectory();
```

Comment:
since this is an overwrite test, you probably don't need to go to the trouble of all this disk work for the initial data, right? Could you just upload a random memory buffer or some random output stream?